### PR TITLE
fix(up.cmd): support `hlx up` in local repo checkout without configured origin

### PR DESCRIPTION
**Pull request by stefan-guggisberg**
## Related Issues

#1189

### CODE REVIEW

File: test/testUpCmd.js [hunk 1]
```diff
@@ -27,6 +27,7 @@ const {
   createTestRoot,
 } = require('./utils.js');
 
+const DemoCommand = require('../src/demo.cmd');
```
Comment (tripodsan) [inline test/testUpCmd.js]: is this really required?

File: src/up.cmd.js [hunk 2]
```diff
@@ -159,20 +159,24 @@ class UpCommand extends BuildCommand {
...
     // check all local repos
-    const localRepos = await Promise.all(this._localRepos.map(async (repo) => {
+    const localRepos = (await Promise.all(this._localRepos.map(async (repo) => {
       const repoPath = path.resolve(this.directory, repo);
       if (!await fse.pathExists(path.join(repoPath, '.git'))) {
         throw Error(`Specified --local-repo = ${repo} is not a git repository.`);
       }
       const gitUrl = await GitUtils.getOriginURL(repoPath);
       if (!gitUrl) {
+        if (repo === '.') {
```
Comment (tripodsan) [inline src/up.cmd.js]: this fixes it, but I think the error is a different one, such as the `.` is a new default.
this was changed #913. I think the check for a local repo should be different.

the original idea was that you can specify a local repo that overrides the resolution for git repositories specified in the strains. Before any strains are setup and before the local git has no origin, then the simulator anyways uses the cwd.

so I think that #913 should be fixed in a way, that it sets `.` as default, only if the local repo has an origin.

**Comment by rofe** [comment]
Should we also update `testDemoUp.js` to use `.` as local repo like it would on the command line?

**Comment by stefan-guggisberg** [comment]
> Should we also update testDemoUp.js to use . as local repo like it would on the command line?

The `--locol-repo .` default is applied conditionally. `testDemoUp.js` is using lower-level components (`DemoCommand`, `UpCommand`) and does not cover command line parameter handling and defaults. For a more realistic test case `testDemoUp.js` should be rewritten using the `CLI`, similar to https://github.com/adobe/helix-cli/blob/d0d73880f13e0aba4444ab6311143822377dd5e4/test/testUpCli.js#L171-L177

**Pull request merged**